app/app.py

```diff
@@ -84,99 +88,141 @@ def get_intersect_features(query_geom, grid, idx):
     # reduce intersection feature set with rtree (this tests polygon bbox intersection)
     for intersect_pos in idx.intersection(query_geom.bounds):
 
-        grid_id = list(grid["FeatureIndex"].keys())[intersect_pos]
-        poly = shape(grid["FeatureIndex"][grid_id]["geometry"])
+        grid_id = list(grid.keys())[intersect_pos]
+        poly = shape(grid[grid_id]["geometry"])
 
         # check if poly actually interesects with request geom
         if query_geom.intersects(poly):
-            ids.append(grid["FeatureIndex"][grid_id]["properties"]["id"])
+            ids.append(grid[grid_id]["properties"][id_property])
             polys.append(poly)
 
     return ids, polys
 
+def get_flow_keys(flow):
+    '''
+    Bit of harcoding to map the flow to the corresponding dataset property
+    '''
+    if flow == "origin":
+        flow_key_init = "orig_cell_id"
+        flow_key_end = "dest_cell_id"
+    elif flow == "destination":
+        flow_key_init = "dest_cell_id"
+        flow_key_end = "orig_cell_id"
+    else:
+        # this should never happen because we validate the flow param when parsing
+        # the request
+        raise exceptions.ServerError("Unsupported flow specified. Must be either origin (default) or destination.", status_code=500)
+
+    return [flow_key_init, flow_key_end]
+
 
-def get_trip_features(intersect_ids, grid, flow, mode):
+def get_trips(intersect_ids, flow_keys, mode):
     '''
     Given a list of cell ids, extract trip count properties from the source grid data.
     '''
 
-    # aggregate trip count values from grid cells that match a source cell
-    trip_features_lookup = {}
+    # this flow O/D stuff can get confusing, so let's name these list elements
+    flow_key_init = flow_keys[0]
+    flow_key_end = flow_keys[1]
 
-    total_trips = 0
+    # generate a string of single-quoted ids (as if for a SQL `IN ()` statement)
+    intersect_id_string = ', '.join([f"'{id_}'" for id_ in intersect_ids])
 
-    if flow == "origin":
-        flow_key = "orig_cell_ids"
-    elif flow == "destination":
-        flow_key = "dest_cell_ids"
+    # todo: this
```

Comment:
Should we make an gh-issue for this?

Reply:
https://github.com/cityofaustin/transportation-dockless-api/issues/33